tasks/tenant/update-manager-image-in-git/tests/mocks.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# mocks to be injected into task step scripts
+function git() {
+  echo "Mock git called with: $*"
+
+  if [[ "$*" == *"clone"* ]]; then
+    gitRepo=$(awk '{print $NF}' <<< $*)
+    mkdir -p "$gitRepo"/internal-services/manager
+    echo "image: quay.io/konflux-ci/internal-services:old" | tee \
+      "$gitRepo"/internal-services/manager/one.yaml \
+      "$gitRepo"/internal-services/manager/two.yaml \
+      "$gitRepo"/internal-services/manager/three.yaml
+  else
+    # Mock the other git functions to pass
+    : # no-op - do nothing
+  fi
+}
+
+function gh() {
+  echo "Mock gh called with: $*"
+
+  if [[ "$*" == *"base fail"* ]]; then
+    echo Error: mocked failure for pr creation
+    exit 1
```

Comment:
If it mocks the `gh` command failure, I think it should be `return 1` so that it still matters how the script handles the call.